components/style/properties/longhand/background.mako.rs

```diff
@@ -13,15 +13,15 @@ ${helpers.predefined_type("background-color", "CSSColor",
 <%helpers:vector_longhand name="background-image" animatable="False">
     use cssparser::ToCss;
     use std::fmt;
-    use values::specified::Image;
+    use values::specified::image::Image;
```

Comment:
you should reexport `Image` and `Gradient` into `values::foo`. `pub use image::Image` etc.

You probably want to also reexport the others, but perhaps rename `Size` and `EndingShape` (since the term can be ambiguous). E.g `pub use image::Size as GradientSize` etc.